### PR TITLE
Bug 2076793: pkg/operator/upgradebackupcontroller: Pivot from Failing to ReleaseAccepted

### PR DESCRIPTION
**Pull request by wking**
[cvo#683][1], backported [to 4.10.8][3] with [cvo#753][2], pivoted the "I'm waiting for RecentBackup" message from the Failing condition to a new `ReleaseAccepted` condition (`Failing` is now focused on "am I happy with the current release reconciliation?" and now excludes "I'm concerned about the requested desired target release, which I haven't accepted yet").  Catch up by looking at the new condition.

We don't need to look at `Failing` anymore, because the cluster-version has already made the pivot to `ReleaseAccepted`, and since the cluster-version operator is the first to update, we don't need to worry about newer etcd operators vs. older cluster-version operators.

[1]: https://github.com/openshift/cluster-version-operator/pull/683/commits/7221c93dfcd19f6c1f46614cec105ba36be2f61e#diff-db97dc5ae62ef3b4c4dabfdc43cbc3ae0ceae9b58748cadebde679e32b5bec2bR160
[2]: https://github.com/openshift/cluster-version-operator/pull/753/files#diff-db97dc5ae62ef3b4c4dabfdc43cbc3ae0ceae9b58748cadebde679e32b5bec2bR160
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=2064991#c7

### CODE REVIEW

File: pkg/operator/upgradebackupcontroller/upgradebackupcontroller_test.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func Test_ensureRecentBackup(t *testing.T) {
 				u.FakePod("cluster-backup", u.WithPodStatus(corev1.PodFailed), u.WithCreationTimestamp(nowMinusDuration(failedPodBackoffDuration))),
 			},
 			clusterversionCondition: configv1.ClusterOperatorStatusCondition{
-				Type:    "Failing",
+				Type:    "ReleaseAccepted",
 				Status:  configv1.ConditionTrue,
```
Comment (LalatenduMohanty) [inline pkg/operator/upgradebackupcontroller/upgradebackupcontroller_test.go:41]:
```
Type:    "Failing",
Status:  configv1.ConditionTrue,
``` 
is different than  , so not sure why this change is done. 
```
Type:    "ReleaseAccepted",
Status:  configv1.ConditionTrue,
```

Comment (LalatenduMohanty) [reply pkg/operator/upgradebackupcontroller/upgradebackupcontroller_test.go:41]: I think I got it now.

**Comment by wking** [comment]
[`KubeContainerWaiting` and similar for marketplace pods][1] is unrelated.

/retest required

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/799/pull-ci-openshift-cluster-etcd-operator-master-e2e-aws-single-node/1517595688383287296

**Comment by EmilyM1** [comment]
Condition swaps look good to me

**Comment by wking** [comment]
[single-node failed to install][1]:

/retest-required

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/799/pull-ci-openshift-cluster-etcd-operator-master-e2e-aws-single-node/1517636070412390400
